vault/data_source_address.go

```diff
@@ -0,0 +1,34 @@
+package vault
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func addressDataSource() *schema.Resource {
+	return &schema.Resource{
+		Read: ReadWrapper(addressDataSourceRead),
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "URL of the root of the target Vault server.",
+			},
+		},
+	}
+}
+
+func addressDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return e
+	}
+
+	address := client.Address()
+
+	d.SetId(address)
```

Comment:
I don't think the `id` has any importance for this data source, but didn't come up with anything else either.